span.go

```diff
@@ -78,8 +78,14 @@ func (r *spanS) FinishWithOptions(opts ot.FinishOptions) {
 
 	r.Duration = duration
 	if !r.context.Suppressed {
-		r.tracer.recorder.RecordSpan(r)
-		r.sendOpenTracingLogRecords()
+		if sensor.Agent().Ready() {
+			r.tracer.recorder.RecordSpan(r)
+			r.sendOpenTracingLogRecords()
```

Comment:
It looks like `r.sendOpenTracingLogRecords()` is called regardless of the if statement, no?
If it happens always, maybe it can be called only once outside of the if statement? Or am I missing something?

Reply:
Done